pkg/status/conditions.go

```diff
@@ -0,0 +1,32 @@
+package status
+
+import (
+	"github.com/kyma-project/lifecycle-manager/api/v1beta1"
+	apimeta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SetConditions checks if all required conditions do exists and removes deprecated Conditions.
+// If one or many Conditions are missing, they will be created and `true` will be returned
+// to determine that an update is needed.
+// If all required conditions exist, they will be set to `False`.
+func SetConditions(kyma *v1beta1.Kyma, watcherEnabled bool) bool {
+	conditionUpdateNeeded := false
+
+	existingConditions := kyma.Status.Conditions
+	// Remove deprecated `Ready` condition
+	if deprecatedCondition := apimeta.FindStatusCondition(existingConditions,
+		string(v1beta1.DeprecatedConditionReady)); deprecatedCondition != nil {
+		apimeta.RemoveStatusCondition(&existingConditions, string(v1beta1.DeprecatedConditionReady))
```

Comment:
the `existingConditions` do not point to the original condition, when you do the assignment `existingConditions := kyma.Status.Conditions`,  the value gets copied, so the remove here will not remove actual `kyma.Status.Conditions`. Please also write a test for it after the fix.